src/services/fs/nfs.rs

```diff
@@ -0,0 +1,740 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Async filesystem primitives.
+//!
+//! This is a big fork of [`async-fs`]. We use [`nuclei`] to support io-uring.
+//!
+//! [`async-fs`]: https://docs.rs/async-fs
+//! [`nuclei`]: https://docs.rs/nuclei
+//!
+//! # Implementation
+//!
+//! This crate uses [`blocking`] to offload blocking I/O onto a thread pool.
+//!
+//! [`blocking`]: https://docs.rs/blocking
+//!
+//! # Examples
+//!
+//! Create a new file and write some bytes to it:
+//!
+//! ```no_run
+//! use opendal::services::fs::nfs::File;
+//! use futures::io::AsyncWriteExt;
+//!
+//! # futures::executor::block_on(async {
+//! let mut file = File::create("a.txt").await?;
+//! file.write_all(b"Hello, world!").await?;
+//! file.flush().await?;
+//! # std::io::Result::Ok(()) });
+//! ```
+
+#![warn(missing_docs)]
+
+use std::fmt;
+use std::io::{self, SeekFrom};
+use std::path::Path;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::{Context, Poll};
+
+use blocking::Unblock;
+use futures::io::{AsyncRead, AsyncSeek, AsyncWrite, AsyncWriteExt};
+use futures::lock::Mutex;
+use futures::{future::Future, ready};
+use nuclei::spawn_blocking;
```

Comment:
nuclei are cool, but I have concerns about their maintenance:

In its dependence:

```
# Other backends
[target.'cfg(target_os = "linux")'.dependencies]
iou = { version = "0.0.0-ringbahn.1", optional = true }
uring-sys = { version = "0.6.1", optional = true }
```

Reply:
Yes, we can try to help improve it with `io-uring`